Use typed ioctls on Solaris

AttilaFueloep opened the pull request:
Solaris go has no syscall.SYS_IOCTL, we have to use typed ioctls instead. This fixes the ```undefined: syscall.SYS_IOCTL``` errors I'm seeing on SmartOS with pksrc 16Q4 go 2.7.4.  I'm wondering on what OS and go combination #7 worked.

See https://github.com/golang/go/issues/12574 and https://github.com/golang/sys/commit/584c5fee74f2fb64acbbd2fc9d1db29c4237d311

gfrey commented on term_unix_ioctl.go:
```diff
@@ -0,0 +1,58 @@
+		uintptr(unsafe.Pointer(termios)))
+
+	if err != 0 {
+		return nil, errors.New(err.Error())
```
This `errors.New(err.Error())` is unnecessary (get error string to create a new error). Same thing in line35 and line55. But minor issue.

larzconwell replied:
I'll probably do a little clean up after I merge.

AttilaFueloep replied:
Yes, @gfrey,  you are right.  Errno implements the error interface, so we can simply return err. We already made sure not to return a zero Errno here. Sorry for missing that.

larzconwell commented:
Thank you for testing @gfrey!! I'll look at this in more detail later tonight.

larzconwell commented on term_solaris_ioctl.go:
```diff
@@ -0,0 +1,37 @@
+// +build solaris
+
+// Copyright 2017 Attila Fülöp <attila@fueloep.org>
```
There's been a few people now who've contributed now and haven't left their name, instead of individual copyright lines in each file, do you think it would be fine to instead place your name in a `CONTRIBUTORS` file?

AttilaFueloep replied:
Sure, that's perfectly fine by me.

AttilaFueloep commented:
I've removed my Copyrights and changed the error handling  in https://github.com/Bowery/prompt/pull/8/commits/5634d832fcee7f5875bb28c9910802ce90395595. Better now?

larzconwell commented:
Yes thank you! Once I get off work I will merge this in(and clean it up a little) and add you and the rest of the contributors to `CONTRIBUTORS`.

AttilaFueloep commented:
Thanks for merging!
